src/idl_gen_rust.cpp

```diff
@@ -2472,12 +2587,45 @@ class RustGenerator : public BaseGenerator {
       code_ += "  }\n";
       // Setter.
       if (IsStruct(field.value.type)) {
-        code_.SetValue("FIELD_SIZE",
-                       NumToString(field.value.type.struct_def->bytesize));
+        code_.SetValue("FIELD_SIZE", NumToString(InlineSize(field.value.type)));
         code_ += "  pub fn set_{{FIELD_NAME}}(&mut self, x: &{{FIELD_TYPE}}) {";
         code_ +=
             "    self.0[{{FIELD_OFFSET}}..{{FIELD_OFFSET}}+{{FIELD_SIZE}}]"
             ".copy_from_slice(&x.0)";
+      } else if (IsArray(field.value.type)) {
+        if (GetFullType(field.value.type) == ftArrayOfBuiltin) {
+          code_.SetValue("ARRAY_ITEM",
+                         GetTypeGet(field.value.type.VectorType()));
+          code_.SetValue(
+              "ARRAY_ITEM_SIZE",
+              NumToString(InlineSize(field.value.type.VectorType())));
+          code_ +=
+              "  pub fn set_{{FIELD_NAME}}(&mut self, items: &{{FIELD_TYPE}}) "
+              "{";
+          code_ += "    for (i, item) in items.iter().enumerate() {";
```

Comment:
I think Rust builtin arrays are contiguous so you can copynonoverlapping without a for loop

Reply:
Yeah, but I had to convert all items to little endian. Do you think we should split this into 2 cases based on the machines endianity?

Reply:
Oh, I see, you're right... 2 cases are fine if you want to but I'm okay with this then